activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -193,6 +205,80 @@ def wait_poll(timeout)
         end
       end
 
+      # Adds the ability to turn a basic fair FIFO queue into one
+      # biased to some thread.
+      module BiasableQueue # :nodoc:
+        class BiasedConditionVariable
```

Comment:
Is this something we could upstream to concurrent-ruby, and reduce the patch size? /cc @jdantonio